python/example_code/sesv2/newsletter.py

```diff
@@ -0,0 +1,312 @@
+                                "Text": {"Data": welcome_text},
+                                "Html": {"Data": welcome_html},
+                            },
+                        }
+                    },
+                )
+                print(f"Welcome email sent to '{email}'.")
+                # snippet-end:[python.example_code.sesv2.SendEmail.simple]
+                if self.sleep:
+                    # 1 email per second in sandbox mode, remove in production.
+                    sleep(1.1)
+            except ClientError as e:
+                # If the contact already exists, skip and proceed
+                if e.response["Error"]["Code"] == "AlreadyExistsException":
+                    print(f"Contact with email '{email}' already exists. Skipping...")
+                else:
+                    raise e
+            # snippet-end:[python.example_code.sesv2.CreateContact]
+
+    def send_coupon_newsletter(self):
+        """
+        Sends the coupon newsletter to the subscribers.
+        """
+        # Get the list of contacts
+        # snippet-start:[python.example_code.sesv2.ListContacts]
+        try:
+            contacts_response = self.ses_client.list_contacts(
+                ContactListName=CONTACT_LIST_NAME
+            )
+        except ClientError as e:
+            if e.response["Error"]["Code"] == "NotFoundException":
+                print(f"Contact list '{CONTACT_LIST_NAME}' does not exist.")
+                return
+            else:
+                raise e
+        # snippet-end:[python.example_code.sesv2.ListContacts]
+
+        # Send the coupon newsletter to each contact
+        coupon_items = load_file_content("sample_coupons.json")
+
+        for contact in contacts_response["Contacts"]:
+            email_address = contact["EmailAddress"]
+            try:
+                # snippet-start:[python.example_code.sesv2.SendEmail.template]
+                send = self.ses_client.send_email(
+                    FromEmailAddress=self.verified_email,
+                    Destination={"ToAddresses": [email_address]},
+                    Content={
+                        "Template": {
+                            "TemplateName": TEMPLATE_NAME,
+                            "TemplateData": coupon_items,
+                        }
+                    },
+                    ListManagementOptions={"ContactListName": CONTACT_LIST_NAME},
+                )
+                # snippet-end:[python.example_code.sesv2.SendEmail.template]
+                print(f"Newsletter sent to '{email_address}'.")
+                print("Debug: ", send)
+                if self.sleep:
+                    # 1 email per second in sandbox mode, remove in production.
+                    sleep(1.1)
+            except ClientError as e:
+                print_error(e)
+
+    def monitor_and_review(self):
+        """
+        Provides instructions for monitoring sending activity in the AWS console.
+        """
+        print(
```

Comment:
Would this look nicer in a multi-line print like the intro?

Reply:
Done